src/js/components/Select/doc.js

```diff
@@ -26,6 +26,13 @@ export const doc = Select => {
     closeOnChange: PropTypes.bool
       .description('Wether to close the drop when a selection is made.')
       .defaultValue(true),
+    clearValueLabel: PropTypes.string
+      .description('Label for the clear selection item')
+      .defaultValue('Clear selection'),
+    clearValuePosition: PropTypes.oneOf(['bottom', 'top']).description(
+      `Add a clear value item add the top or at the bottom of the container.
+      By default no unselect option is present.`,
+    ),
```

Comment:
Cards on the table, I think that Select is one of the most complicated components in Grommet, lots of props and options it can get overwhelming for users.
In the case above, I feel like we are adding two new props but one of them wouldn't make sense without the other, hence, IMO it should be only one object prop that will gather all the `clear` value attributes, so if the prop is undefined, the clear option wouldn't be available.
So something like this:
```
<Select clear={{label: `Clear`, value: 'top'}} ... />
```

We can even take it one step forward and combine the `renderClearValue` prop with it as well, something like:
```
<Select clear={{label: `Clear`, value: 'top', action: ()=> ()}} ... />
```

I would love to hear your thoughts here, @markjackson02  welcome to join as well.

Reply:
yeah great suggestion. i think it totally makes sense and i should do that for render prop as well.

Reply:
i agree!